src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -118,20 +119,38 @@ def migrate(self, daemon_name=None, realm_name=None, zonegroup_name=None, zone_n
     # pylint: disable=W0102,W0613
     def setup_multisite_replication(self, daemon_name=None, realm_name=None, zonegroup_name=None,
                                     zonegroup_endpoints=None, zone_name=None, zone_endpoints=None,
-                                    username=None, cluster_fsid=None):
-        multisite_instance = RgwMultisite()
+                                    username=None, cluster_fsid=None, replication_zone_name=None,
+                                    cluster_details=None):
+        multisite_instance = RgwMultisiteAutomation()
         result = multisite_instance.setup_multisite_replication(realm_name, zonegroup_name,
                                                                 zonegroup_endpoints, zone_name,
                                                                 zone_endpoints, username,
-                                                                cluster_fsid)
+                                                                cluster_fsid,
+                                                                replication_zone_name,
+                                                                cluster_details)
         return result
 
     @RESTController.Collection(method='PUT', path='/setup-rgw-credentials')
     @allow_empty_body
     # pylint: disable=W0102,W0613
     def restart_rgw_daemons_and_set_credentials(self):
         rgw_service_manager_instance = RgwServiceManager()
-        result = rgw_service_manager_instance.restart_rgw_daemons_and_set_credentials()
+        result = rgw_service_manager_instance.configure_rgw_credentials()
+        return result
+
+    @RESTController.Collection(method='GET', path='/available-ports')
+    @allow_empty_body
+    # pylint: disable=W0102,W0613
+    def get_available_ports(self):
+        rgw_service_manager_instance = RgwServiceManager()
+        result = rgw_service_manager_instance.find_available_port()
+        return result
```

Comment:
why do we need to find the available port? isn't it the responsibility of cephadm to schedule the daemon to a port which it sees that is free?

Reply:
No, its not being taken care of by cephadm at present, if you run a new rgw service it takes `port 80` by default every time.

Reply:
Probably we should start a conversation with cephadm to improve this behavior. Cephadm is in charge of deploying and managing a service and we can try to improve the behavior by making it smart enough to find an available port for a daemon that it is trying to deploy if the given port is unavailable. Its a worthy conversation to have IMO.